### ISSUE TITLE
HttpStatus error

### ISSUE THREAD

**Issue by key88sf** [rubocop-hq/rubocop#6794]
Rails controller code:
```
class PromotionsController < ApplicationController
  def show
    if user_signed_in?
      redirect_to welcome_path
    else
      redirect_to root_path(utm_source: :pr, utm_medium: :web), status: 301
    end
  end
end
```

Getting the following error:
```
An error occurred while Rails/HttpStatus cop was inspecting app/controllers/promotions_controller.rb:6:6.
To see the complete backtrace run rubocop -d.

1 error occurred:

Errors are usually caused by RuboCop bugs.

Please, report your problems to RuboCop's issue tracker.

https://github.com/rubocop-hq/rubocop/issues

Mention the following information in the issue report:

0.59.2 (using Parser 2.5.1.2, running on ruby 2.6.1 x86_64-linux)

Inspecting 1 file

.

1 file inspected, no offenses detected
```
--------

## Expected behavior

Expected no errors from Hound/Rubocop

## Actual behavior

Appears to crash when the status_code: 301 is added.

## Steps to reproduce the problem

Write a simple controller that redirects with a status code.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
0.59.2 (using Parser 2.5.1.2, running on ruby 2.6.1 x86_64-linux)
```

**Comment by mikegee** [comment]
@key88sf please confirm whether or not this issue is a duplicate of https://github.com/codeclimate/codeclimate-rubocop/issues/143.

**Comment by tejasbubane** [comment]
Cannot reproduce this on master. The version mentioned in issue is slightly old. I went ahead and added related specs.

**Comment by tiev** [comment]
I have the same error on version

`0.63.1 (using Parser 2.6.0.0, running on ruby 2.5.3 x86_64-linux)`

The code line is this:

```ruby
      render json: { error: t("reached_limit") }, status: 403
```

**Comment by tiev** [comment]
Update:

Still has the same error on the newest version.

`0.65.0 (using Parser 2.6.0.0, running on ruby 2.5.3 x86_64-linux)`

I use rubocop 0.65.0 and rubocop-rspec 1.32.0

**Comment by mikegee** [comment]
Please run `rubocop -d` to show the backtrace.

**Comment by tiev** [comment]
Backtrace:

```
Scanning /app/controllers/memberships_controller.rb
uninitialized constant Rack
Did you mean?  Racc
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/rails/http_status.rb:136:in `custom_http_status_code?'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/rails/http_status.rb:109:in `offensive?'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/rails/http_status.rb:61:in `block in on_send'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/rails/http_status.rb:46:in `http_status'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/rails/http_status.rb:56:in `on_send'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/commissioner.rb:58:in `block (2 levels) in trigger_responding_cops'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/commissioner.rb:106:in `with_cop_error_handling'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/commissioner.rb:57:in `block in trigger_responding_cops'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/commissioner.rb:56:in `each'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/commissioner.rb:56:in `trigger_responding_cops'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/commissioner.rb:34:in `block (2 levels) in <class:Commissioner>'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/ast/traversal.rb:141:in `on_if'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/commissioner.rb:35:in `block (2 levels) in <class:Commissioner>'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/ast/traversal.rb:100:in `on_def'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/commissioner.rb:35:in `block (2 levels) in <class:Commissioner>'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/ast/traversal.rb:50:in `block in on_begin'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/ast/traversal.rb:50:in `each'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/ast/traversal.rb:50:in `on_begin'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/commissioner.rb:35:in `block (2 levels) in <class:Commissioner>'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/ast/traversal.rb:92:in `on_class'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/commissioner.rb:35:in `block (2 levels) in <class:Commissioner>'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/ast/traversal.rb:13:in `walk'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/commissioner.rb:46:in `investigate'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/team.rb:116:in `investigate'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/team.rb:104:in `offenses'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cop/team.rb:44:in `inspect_file'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/runner.rb:280:in `inspect_file'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/runner.rb:227:in `block in do_inspection_loop'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/runner.rb:259:in `block in iterate_until_no_changes'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/runner.rb:252:in `loop'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/runner.rb:252:in `iterate_until_no_changes'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/runner.rb:223:in `do_inspection_loop'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/runner.rb:126:in `block in file_offenses'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/runner.rb:144:in `file_offense_cache'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/runner.rb:124:in `file_offenses'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/runner.rb:112:in `process_file'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/runner.rb:89:in `block in each_inspected_file'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/runner.rb:86:in `each'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/runner.rb:86:in `reduce'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/runner.rb:86:in `each_inspected_file'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/runner.rb:76:in `inspect_files'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/runner.rb:48:in `run'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cli.rb:174:in `execute_runner'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cli.rb:75:in `execute_runners'
/usr/local/bundle/gems/rubocop-0.65.0/lib/rubocop/cli.rb:47:in `run'
/usr/local/bundle/gems/rubocop-0.65.0/exe/rubocop:13:in `block in <top (required)>'
/usr/local/lib/ruby/2.5.0/benchmark.rb:308:in `realtime'
/usr/local/bundle/gems/rubocop-0.65.0/exe/rubocop:12:in `<top (required)>'
/usr/local/bundle/bin/rubocop:23:in `load'
/usr/local/bundle/bin/rubocop:23:in `<main>'
```

**Comment by tiev** [comment]
Seems we should extract Rails cops to gem `rubocop-rails`, and add dependencies for Rails there.

**Comment by mikegee** [comment]
Just looking at the code to confirm, this backtrace shows a codepath that references `Rack` that isn't conditional on `RACK_LOADED`. I think this issue should be reopened.
